tests/multiphase_flow/free_falling_cyl_cib.cpp

```diff
@@ -0,0 +1,2202 @@
+        const Box<NDIM>& patch_box = patch->getBox();
+        Pointer<CellData<NDIM, int> > tags_data = patch->getPatchData(tag_index);
+        Pointer<CellData<NDIM, double> > ls_data = patch->getPatchData(ls_current_idx);
+
+        for (CellIterator<NDIM> ic(patch_box); ic; ic++)
+        {
+            const hier::Index<NDIM>& i = ic();
+            const double dist_norm = std::abs((*ls_data)(i)-ptr_ls_tagger->d_tag_value);
+
+            if (dist_norm <= ptr_ls_tagger->d_tag_abs_thresh)
+            {
+                (*tags_data)(i) = 1;
+            }
+        }
+    }
+
+    return;
+
+} // callTagLSRefinementCellsCallBackFunction
+
+namespace IBAMR
+{
+/*!
+ * \brief Class RigidBodyKinematics provides definition for base ConstraintIBKinematics class.
+ */
+class RigidBodyKinematics : public ConstraintIBKinematics
+{
+public:
+    /*!
+     * \brief Constructor.
+     */
+    RigidBodyKinematics(const std::string& object_name,
+                        SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db,
+                        IBTK::LDataManager* l_data_manager,
+                        SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > patch_hierarchy,
+                        bool register_for_restart = true);
+
+    /*!
+     * \brief Destructor.
+     */
+    virtual ~RigidBodyKinematics();
+
+    /*!
+     * \brief Set kinematics velocity at new time for rigid body.
+     */
+    virtual void setKinematicsVelocity(const double time,
+                                       const std::vector<double>& incremented_angle_from_reference_axis,
+                                       const std::vector<double>& center_of_mass,
+                                       const std::vector<double>& tagged_pt_position);
+
+    /*!
+     * \brief Get the kinematics velocity at new time for rigid body on the specified level.
+     */
+    virtual const std::vector<std::vector<double> >& getKinematicsVelocity(const int level) const;
+
+    /*!
+     * \brief Set the shape of rigid body at new time on all levels.
+     */
+    virtual void setShape(const double time, const std::vector<double>& incremented_angle_from_reference_axis);
+
+    /*!
+     * \brief Get the shape of rigid body at new time on the specified level.
+     */
+    virtual const std::vector<std::vector<double> >& getShape(const int level) const;
+
+    // No need for database code in this example
+private:
+    /*!
+     * \brief Deleted default ctor.
+     */
+    RigidBodyKinematics();
+
+    /*!
+     * \brief Deleted default copy ctor.
+     */
+    RigidBodyKinematics(const RigidBodyKinematics& from);
+
+    /*!
+     * \brief Deleted default assignment.
+     */
+    RigidBodyKinematics& operator=(const RigidBodyKinematics& that);
+
+    // No need for restart code in this example
+
+    /*!
+     * \brief Set rigid body velocity.
+     */
+    void setRigidBodySpecificVelocity(const double time,
+          
```

Comment:
Here is the second half of the trick to get the absolute path to an input file.

Reply:
I don't follow why this trick is needed. Why can't we dump the input files in the right place automatically?